packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+          ),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(createResultSetName, 'Mutation'));
+
+      // user cannot create a record with public protected field
+      await expect(
+        async () => await user1TodoHelper.create(createResultSetName, { ...todo, publicContent: 'Public Content' }),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(createResultSetName, 'Mutation'));
+
+      // Create a record with non-public fields which is allowed, so we can test the update and delete operations.
+      const createResult = await user1TodoHelper.create(createResultSetName, { ...todo, authors: [userName1] }, ownerResultSet);
+      checkOperationResult(createResult, { ...todo, authors: [userName1], privateContent: null }, createResultSetName);
+      todo['authors'] = [userName1];
+
+      const publicFieldSet = `
+        customId
+        publicContent
+      `;
+      // owner cannot update a record with public protected field
+      await expect(
+        async () =>
+          await user1TodoHelper.update(
+            updateResultSetName,
+            { customId: todo['customId'], publicContent: 'Public Content' },
+            publicFieldSet,
+          ),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record to re-assign ownership
+      const ownerFieldSet = `
+        customId
+        authors
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { customId: todo['customId'], authors: ['user2'] }, ownerFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record with a dynamic owner list protected field
+      const ownersContentFieldSet = `
+        customId
+        ownersContent
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { ...todo, ownersContent: 'Owners Content' }, ownersContentFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // owner cannot read a record with public protected field
+      const getResult1 = await user1TodoHelper.get({ customId: todo['customId'] }, publicFieldSet, false, 'all', 'customId');
+      checkOperationResult(
+        getResult1,
+        { customId: todo['customId'], publicContent: null },
+        `get${modelName}`,
+        false,
+        expectedFieldErrors(['publicContent'], modelName),
+      );
+
+      const listTodosResult1 = await user1TodoHelper.list({}, publicFieldSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['customId'], true, 'customId');
+      checkListResponseErrors(listTodosResult1, expectedFieldErrors(['publicContent'], modelName, false));
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket